web/src/engine/websites/decorators/MangaReaderCMS.ts

```diff
@@ -0,0 +1,189 @@
+import { FetchRequest, FetchCSS } from '../../FetchProvider';
+import { type MangaScraper, Manga, type Chapter, type Page, type MangaPlugin } from '../../providers/MangaPlugin';
+import * as Common from './Common';
+
+function ChapterInfoExtractor(element : HTMLElement) {
+    const anchor: HTMLAnchorElement = element.nodeName.toLowerCase() === 'a' ? element as HTMLAnchorElement : element.querySelector('a');
+    const id = anchor.pathname;
+    const title = element.innerText.replace(/\s*:\s*$/, '').trim();
+    return { id, title };
+}
+
+const queryMangas = [
+    'ul.manga-list li a',
+    'ul.manga-list-text li a.alpha-link',
+    'ul.price-list li a'
+].join(',');
+
+const queryChapters = [
+    'ul.chapters li h3.chapter-title-rtl a',
+    'ul.chapters li h5.chapter-title-rtl a',
+    'ul.chapter-list li a',
+    'div.chapters h3.chapter-title a',
+    'div.chapter-wrapper table td.chapter a',
+    'div.capitulos-list table tr td:first-of-type a',
+    'ul li div.item a.chapter_num',
+    'table.table--manga tbody td.table__chapter a',
+    'ul.chapterszozo li a',
+].join(',');
+
+const queryPages = [
+    'div#all img.img-responsive',
+    'div.text-center img[loading="lazy"]',
+].join(',');
+
+const queryMangaTitle = [
+    'h1.widget-title',
+    'h2.widget-title',
+    'h2.listmanga-header'
+].join(',');
+
+const pathname = '/';
+const DefaultChapterExtractor = ChapterInfoExtractor;
+
+/***************************************************
+ ******** Manga from URL Extraction Methods ********
+ ***************************************************/
+
+/**
+ * An extension method for extracting a single manga from the given {@link url} using the given CSS {@link query}.
+ * The `pathname` of the given {@link url} will be used as identifier for the extracted manga.
+ * When the CSS {@link query} matches a `meta` element, the manga title will be extracted from its `content` attribute, otherwise the `textContent` of the element will be used as manga title.
+ * @param this - A reference to the {@link MangaScraper} instance which will be used as context for this method
+ * @param provider - A reference to the {@link MangaPlugin} which shall be assigned as parent for the extracted manga
+ * @param url - The url from which the manga shall be extracted
+ * @param query - A CSS query to locate the element from which the manga title shall be extracted
+ */
+export async function FetchMangaCSS(this: MangaScraper, provider: MangaPlugin, url: string, query: string = queryMangaTitle): Promise<Manga> {
```

Comment:
This seems to be just a wrapper for `Common.FetchMangaCSS` with a specific default value for query.
Suggestion to not implement this method and just export `QueryMangaTitle`, so it can be used like this:
```typescript
import * as Common from '../decorators/Common';
import * as MangaReaderCMS from '../decorators/MangaReaderCMS';

export default class extends DecoratableMangaPlugin {
    public override async FetchManga() {
        // ...
        await Common.FetchMangaCSS(..., MangaReaderCMS.QueryMangaTitle);
    }
}
```

Reply:
The same may be checked for other methods in this class, if these can be easily replaced by methods from `Common` there is no need for an additional implementation

Reply:
Hum. I'd rather not override fetch methods in websites plugins if i can. I thought the point of decorators would be not to override FetchX methods each time, which increase a bit the volume of code. 

I also think that overriding FetchX methods makes the code a bit harder to read, and we may miss other things added in FetchX in the future (making the code harder to maintain ? )

Moreover doing like you said would not be consistent with already coded decorators (by me or other contributors)

Maybe i can just exports query values, and use them like that ? 

``` @Common.FetchMangaCSS(..., MangaReaderCMS.QueryMangaTitle); ```

Reply:
Would something like that be ok in a website plugin then? 

``` 
@Common.MangaCSS(/^https?:\/\/manga\.fascans\.com\/manga\/[^/]+$/, MangaReader.queryMangaTitle)
@MangaReader.MangasSinglePageCSS() //unique method so its ok
@Common.ChaptersSinglePageCSS(MangaReader.queryChapters, MangaReader.ChapterInfoExtractor)
@Common.PagesSinglePageCSS(MangaReader.queryPages, MangaReader.ChapterPageExtractor)
@Common.ImageAjax()

```

Reply:
Yes, that meets very much what i had in mind when suggesting the change

Reply:
And also yes, the `MangasMultiPageCSS` methods could be replaced by the `Common` decorator, since they are just a wrapper with a pre-defined path